pkg/core/resources/store/pagination_store.go

```diff
@@ -66,9 +66,19 @@ func (p *paginationStore) List(ctx context.Context, list model.ResourceList, opt
 		return err
 	}
 
-	for _, item := range fullList.GetItems() {
-		if opts.Filter(item) {
-			_ = filteredList.AddItem(item)
+	if len(opts.ResourceKeys) > 0 {
+		for _, rk := range opts.ResourceKeys {
+			for _, item := range fullList.GetItems() {
+				if item.GetMeta().GetMesh() == rk.Mesh && item.GetMeta().GetName() == rk.Name {
+					_ = filteredList.AddItem(item)
+				}
+			}
+		}
+	} else {
```

Comment:
same, I think this should be outside of else

Reply:
it seems the solution would be to do
```
for _, item := range fullList.GetItems() {
  _, exists := opts.ResourceKeys[model.MetaToResourceKey(item.GetMeta())]
  if len(opts.ResourceKeys) > 0 && !exists {
    continue
  }
  if !opts.Filter(item) {
    continue
  }
  _ = filteredList.AddItem(item)
}
```
but ok, we don't use both filters at once. We can fix it if we actually need it